tests/test_layers.py

```diff
@@ -1308,788 +1397,807 @@ def _batch_norm_helper(self, data, gamma, beta, scale, center, axis):
+        weights[2] = np.random.uniform(size=weights[2].shape)
+        weights[5] = weights[2]
+        rnn_layer.set_weights(weights)
 
-            x = np.random.uniform(0.1, 1.0, size=(4, 3, 5)).astype(np.float32)
-            inputs = [x]
+        # Test with random bias
+        expected = model.predict(x)
+        onnx_model = keras2onnx.convert_keras(model, model.name)
+        assert runner(onnx_model.graph.name, onnx_model, x, expected)
 
-            expected = model.predict(inputs)
-            onnx_model = keras2onnx.convert_keras(model, model.name)
-            self.assertTrue(run_onnx_runtime(onnx_model.graph.name, onnx_model, inputs, expected, self.model_files))
 
-            input2 = Input(shape=(None, 5))
-            states = Bidirectional(rnn_class(2, return_state=True))(input1)[1:]
-            out = Bidirectional(rnn_class(2, return_sequences=True))(input2, initial_state=states)
-            model = Model([input1, input2], out)
-            inputs = [x, x]
+def test_Bidirectional_with_initial_states(runner):
+    for rnn_class in [SimpleRNN, GRU, LSTM]:
+        input1 = Input(shape=(None, 5))
+        states = Bidirectional(rnn_class(2, return_state=True))(input1)
+        model = Model(input1, states)
 
-            expected = model.predict(inputs)
-            onnx_model = keras2onnx.convert_keras(model, model.name)
-            self.assertTrue(run_onnx_runtime(onnx_model.graph.name, onnx_model, inputs, expected, self.model_files, atol=1e-5))
+        x = np.random.uniform(0.1, 1.0, size=(4, 3, 5)).astype(np.float32)
+        inputs = [x]
 
-    # Bidirectional LSTM with seq_length = None
-    @unittest.skipIf(get_maximum_opset_supported() < 5,
-                     "None seq_length Bidirectional LSTM is not supported before opset 5.")
-    def test_Bidirectional_seqlen_none(self):
-        for rnn_class in [SimpleRNN, GRU, LSTM]:
-            model = Sequential()
-            model.add(Embedding(39, 128))
-            model.add(Bidirectional(rnn_class(256, input_shape=(None, 32), return_sequences=True)))
-            model.add(Dense(44))
+        expected = model.predict(inputs)
+        onnx_model = keras2onnx.convert_keras(model, model.name)
+        assert runner(onnx_model.graph.name, onnx_model, inputs, expected)
 
-            onnx_model = keras2onnx.convert_keras(model, model.name)
-            for batch in [1, 4]:
-                x = np.random.rand(batch, 50).astype(np.float32)
-                expected = model.predict(x)
-                self.assertTrue(run_onnx_runtime(onnx_model.graph.name, onnx_model, x, expected, self.model_files))
+        input2 = Input(shape=(None, 5))
+        states = Bidirectional(rnn_class(2, return_state=True))(input1)[1:]
+        out = Bidirectional(rnn_class(2, return_sequences=True))(input2, initial_state=states)
+        model = Model([input1, input2], out)
+        inputs = [x, x]
 
-    def test_rnn_state_passing(self):
-        K.clear_session()
```

Comment:
This is being run for all tests now in the `runner` fixture. It helps avoid difficult to reproduce errors that occur due to resource leaking in the tests.